concourse/client/api.py

```diff
@@ -221,9 +221,24 @@ def build_events(self, build_id):
     @ensure_annotations
     def trigger_resource_check(self, pipeline_name: str, resource_name: str):
```

Comment:
I would have preferred (also, but not also) to decrease nesting, to specify the retries here. this way, it would:

a) be explicit that retries are done
b) configurable (a caller could decide not to do retries)

Also, did I mention the reduced nesting?

-> init the `retries` as default value, like so:
```
trigger_resource_check(...... retries=5) # or retries_left?
```